components/cam/src/dynamics/se/restart_dynamics.F90

```diff
@@ -637,6 +642,9 @@ subroutine read_restart_dynamics (File, dyn_in, dyn_out, NLFileName)
     endif
 #endif
 
+    do ie = 1,nelemd
+       elem(ie)%state%Qdp = 0
```

Comment:
If without this there are nans in printstate, does it mean Qdp should have been init-ed with Q*dp (Q and dp i assume are read from restart)?

Reply:
It's a separate issue as to whether Qdp should be set to Q*dp.

To be clear, restart only has values for one of Qdp's two time slots.

printstate is seeing NaNs because it is sometimes looking at the wrong time level. Or maybe it's looking at both Qdp time levels; printstate is a bit of a mess.